src/Http/Http.Extensions/test/RequestDelegateGenerator/Baselines/MapAction_MultipleStringParam_StringReturn.generated.txt

```diff
@@ -0,0 +1,196 @@
+    using RequestDelegateFactoryFunc = System.Func<System.Delegate, Microsoft.AspNetCore.Http.RequestDelegateFactoryOptions, Microsoft.AspNetCore.Http.RequestDelegateMetadataResult?, Microsoft.AspNetCore.Http.RequestDelegateResult>;
+
+    file static class GeneratedRouteBuilderExtensionsCore
+    {
+
+        private static readonly Dictionary<(string, int), (MetadataPopulator, RequestDelegateFactoryFunc)> map = new()
+        {
+            [(@"TestMapActions.cs", 15)] = (
+               (methodInfo, options) =>
+                {
+                    Debug.Assert(options?.EndpointBuilder != null, "EndpointBuilder not found.");
+                    options.EndpointBuilder.Metadata.Add(new SourceKey(@"TestMapActions.cs", 15));
+                    return new RequestDelegateMetadataResult { EndpointMetadata = options.EndpointBuilder.Metadata.AsReadOnly() };
+                },
+                (del, options, inferredMetadataResult) =>
+                {
+                    var handler = (System.Func<string, string, string>)del;
+                    EndpointFilterDelegate? filteredInvocation = null;
+
+                    if (options?.EndpointBuilder?.FilterFactories.Count > 0)
+                    {
+                        filteredInvocation = GeneratedRouteBuilderExtensionsCore.BuildFilterDelegate(ic =>
+                        {
+                            if (ic.HttpContext.Response.StatusCode == 400)
+                            {
+                                return ValueTask.FromResult<object?>(Results.Empty);
+                            }
+                            return ValueTask.FromResult<object?>(handler(ic.GetArgument<string>(0), ic.GetArgument<string>(1)));
+                        },
+                        options.EndpointBuilder,
+                        handler.Method);
+                    }
+
+                    Task RequestHandler(HttpContext httpContext)
+                    {
+                        // Endpoint Parameter: p1 (Type = string, IsOptional = False, Source = Query)
+                        var p1_raw = httpContext.Request.Query["p1"];
+                        if (StringValues.IsNullOrEmpty(p1_raw) && true)
+                        {
+                            httpContext.Response.StatusCode = 400;
+                            return Task.CompletedTask;
+                        }
+                        var p1_local = p1_raw.ToString();
+
+                        // Endpoint Parameter: p2 (Type = string, IsOptional = False, Source = Query)
+                        var p2_raw = httpContext.Request.Query["p2"];
+                        if (StringValues.IsNullOrEmpty(p2_raw) && true)
+                        {
+                            httpContext.Response.StatusCode = 400;
+                            return Task.CompletedTask;
+                        }
+                        var p2_local = p2_raw.ToString();
+
+
```

Comment:
Extra whitespace 😢

Reply:
I'll see if I can figure out where that extra one is coming from. That said I think we'll drive ourselves nuts trying to track down errant new lines as we iterate on this thing. I think it would be better to just do a blanket code-format before we spit the code out at the very end rather than trying to perfect line feeds and indentation.

Reply:
To give you a concrete example of what sweating a bit of whitespace is going to do:

```csharp
        for (var parameterIndex = 0; parameterIndex < endpoint.Parameters.Length; parameterIndex++)
        {
            var parameter = endpoint.Parameters[parameterIndex];

            var parameterPreparationCode = parameter switch
            {
                {
                    Source: EndpointParameterSource.SpecialType
                } => parameter.EmitSpecialParameterPreparation(),
                {
                    Source: EndpointParameterSource.Query,
                } => parameter.EmitQueryParameterPreparation(),
                _ => throw new Exception("Unreachable!")
            };

            // To avoid having two newlines after the block of parameter handling code.
            if (parameterIndex < endpoint.Parameters.Length - 1)
            {
                parameterPreparationBuilder.AppendLine(parameterPreparationCode);
            }
            else
            {
                parameterPreparationBuilder.Append(parameterPreparationCode);
            }
        }
```

The last if block is just noise from a maintainability point of view.